rcl/include/rcl/graph.h

```diff
@@ -581,6 +582,95 @@ rcl_count_subscribers(
   const char * topic_name,
   size_t * count);
 
+/// Wait for there to be a specified number of publishers on a given topic.
+/**
+ * The `node` parameter must point to a valid node.
+ *
+ * The `allocator` parameter must point to a valid allocator.
+ *
+ * The `topic_name` parameter must not be `NULL`, and must not be an empty string.
+ * It should also follow the topic name rules.
+ *
+ * This function blocks and will return when the number of publishers for `topic_name`
+ * is greater than or equal to the `count` parameter, or the specified `timeout` is reached.
```

Comment:
I don't know where, but it might be worth mentioning here, that the node's graph guard condition cannot be used concurrently by any other wait sets, as there is only one and guard conditions can be used with at most one wait set at a time.

Reply:
Added note about it: 0b94ee5